File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>
#include <functional>

using namespace std;

int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    function<void(int,int)> dfs = [&](int idx, int total) {
        if (idx == numbers.size()) {
            if (total == target) {
                answer++;
            }
            return;
        }
        
        dfs(idx+1, total+numbers[idx]);
        dfs(idx+1, total-numbers[idx]);
    };
    dfs(0,0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    
    var answer = 0
    
    func dfs(_ st: Int,_ total: Int) {
        if st == numbers.count {
            if total == target {
                answer += 1
            }
            return
        }
        
        dfs(st+1,total+numbers[st])
        dfs(st+1,total-numbers[st])
    }
    
    dfs(1,numbers[0])
    dfs(1,-numbers[0])
    
    return answer
}